### PR TITLE
Default Timeout for Locator Assertions

### PR DESCRIPTION
**Pull request by uchagani**
I took a shot at implementing #842.  I'm not sure if this is the proper approach but I'm hoping it can at least get the ball rolling.  Happy to make changes with any guidance you can provide.

### CODE REVIEW

**Comment by uchagani** [comment]
@yury-s @mxschmitt I'm not sure why some of these tests are failing.  Doesn't seem to have anything to do with this PR.  I tried to kick off a re-run of the jobs but I'm guessing permissions issues don't allow that.  Would you be able to kick off a re-run to see if these are flaky tests?

**Comment by uchagani** [comment]
I found some bugs i'm working through.  closing for now.

**Comment by uchagani** [comment]
@mxschmitt it seems the node version has a way to configure timeouts via a config: https://playwright.dev/docs/test-timeouts#set-expect-timeout-in-the-config.  If my understanding is correct, this would be similar to a global LocatorAssertions timeout, correct?

**Comment by mxschmitt** [comment]
Yeah but it affects Python and .NET too. Please file an issue in https://github.com/microsoft/playwright

**Comment by uchagani** [comment]
ah ok, makes sense.

File: playwright/src/test/java/com/microsoft/playwright/TestLocatorAssertions.java [hunk 1]
```diff
@@ -165,6 +165,66 @@ void hasTextWTextArrayFailOnNotEmpty() {
...
+    page.evaluate("setTimeout(() => {\n" +
+      "  div.innerHTML = \"<p>Text 1</p><p>Text 2</p>\";\n" +
+      "}, 100);");
+    try {
+      // Should normalize whitespace.
+      assertThat(locator).hasText(new String[] {"Text  1", "Text   2"});
+      fail("did not throw");
+    } catch (AssertionFailedError e) {
+      assertEquals("[Text  1, Text   2]", e.getExpected().getStringRepresentation());
+      assertEquals("null", e.getActual().getStringRepresentation());
```
Comment (uchagani) [inline playwright/src/test/java/com/microsoft/playwright/TestLocatorAssertions.java:190]: @yury-s @mxschmitt I could use some help here.  `e.getExpected()`returns null when the default timeout is set to 0.  Any value great than 0 will work as expected but i'm not sure why that is.

**Comment by uchagani** [comment]
Re-opening this PR as @pavelfeldman said it was okay to do this in just Java.  I could use some help though (see comment above).

File: playwright/src/test/java/com/microsoft/playwright/TestLocatorAssertions.java (hunk 1 above)
Comment (uchagani) [reply playwright/src/test/java/com/microsoft/playwright/TestLocatorAssertions.java:190]: `getExpected()` and `getActual()` both return null:

image

**Comment by uchagani** [comment]
I'll start working on the upstream part this weekend

**Pull request closed without merging**